### ISSUE TITLE
HDP 2.4 on HiBench 6.0 mapreduce.home

### ISSUE THREAD

**Issue by jiaqianjing** [intel-hadoop/HiBench#361]
When I updata new version 6.0, I can't set up my `mapreduce.home`:
`AssertionError: The pattern /usr/hdp/2.4.0.0-169/hadoop/hadoop-mapreduce-examples.jar matches no files, please set `hibench.hadoop.examples.jar` manually
/home/hdfs/HiBench-6.0/bin/functions/workload-functions.sh: line 38: .: filename argument required
.: usage: . filename [arguments]`

**Comment by harschware** [comment]
@carsonwang  Yep, I'm noticing this too.  This is what I get when I try to run run-all.sh.  I have a theory that you can get around it by putting the 00-default-properties.conf from the 5.0 branch into the conf directory.  I know that will get me  past the issue, but there may be more issues.  I have not been able to run 6.0 successful yet.

```
[builder@plato-nid00030 bin]$ ./run-all.sh 
Prepare micro.sleep ...
Exec script: /home/users/builder/cactus-test-framework/cactus-test-framework/workspace/plato-cactus-hibench_AQA-1256/hibench/bin/workloads/micro/sleep/prepare/prepare.sh
patching args=
Parsing conf: /home/users/builder/cactus-test-framework/cactus-test-framework/workspace/plato-cactus-hibench_AQA-1256/hibench/conf/hadoop.conf
Parsing conf: /home/users/builder/cactus-test-framework/cactus-test-framework/workspace/plato-cactus-hibench_AQA-1256/hibench/conf/hibench.conf
Parsing conf: /home/users/builder/cactus-test-framework/cactus-test-framework/workspace/plato-cactus-hibench_AQA-1256/hibench/conf/spark.conf
Parsing conf: /home/users/builder/cactus-test-framework/cactus-test-framework/workspace/plato-cactus-hibench_AQA-1256/hibench/conf/workloads/micro/sleep.conf
probe sleep jar: /usr/hdp/current/hadoop-mapreduce-client/hadoop-mapreduce-client-jobclient.jar
Traceback (most recent call last):
  File "/home/users/builder/cactus-test-framework/cactus-test-framework/workspace/plato-cactus-hibench_AQA-1256/hibench/bin/functions/load_config.py", line 723, in <module>
    load_config(conf_root, workload_configFile, workload_folder, patching_config)
  File "/home/users/builder/cactus-test-framework/cactus-test-framework/workspace/plato-cactus-hibench_AQA-1256/hibench/bin/functions/load_config.py", line 208, in load_config
    generate_optional_value()
  File "/home/users/builder/cactus-test-framework/cactus-test-framework/workspace/plato-cactus-hibench_AQA-1256/hibench/bin/functions/load_config.py", line 646, in generate_optional_value
    probe_masters_slaves_hostnames()
  File "/home/users/builder/cactus-test-framework/cactus-test-framework/workspace/plato-cactus-hibench_AQA-1256/hibench/bin/functions/load_config.py", line 581, in probe_masters_slaves_hostnames
    HibenchConf['hibench.masters.hostnames'].split() +
KeyError: 'hibench.masters.hostnames'
/home/users/builder/cactus-test-framework/cactus-test-framework/workspace/plato-cactus-hibench_AQA-1256/hibench/bin/functions/workload-functions.sh: line 38: .: filename argument required
.: usage: . filename [arguments]
start PrepareSleep bench
Prepare sleep: nothing to do
finish PrepareSleep bench
Run micro/sleep/hadoop
Exec script: /home/users/builder/cactus-test-framework/cactus-test-framework/workspace/plato-cactus-hibench_AQA-1256/hibench/bin/workloads/micro/sleep/hadoop/run.sh
patching args=
Parsing conf: /home/users/builder/cactus-test-framework/cactus-test-framework/workspace/plato-cactus-hibench_AQA-1256/hibench/conf/hadoop.conf
Parsing conf: /home/users/builder/cactus-test-framework/cactus-test-framework/workspace/plato-cactus-hibench_AQA-1256/hibench/conf/hibench.conf
Parsing conf: /home/users/builder/cactus-test-framework/cactus-test-framework/workspace/plato-cactus-hibench_AQA-1256/hibench/conf/spark.conf
Parsing conf: /home/users/builder/cactus-test-framework/cactus-test-framework/workspace/plato-cactus-hibench_AQA-1256/hibench/conf/workloads/micro/sleep.conf
probe sleep jar: /usr/hdp/current/hadoop-mapreduce-client/hadoop-mapreduce-client-jobclient.jar
Traceback (most recent call last):
  File "/home/users/builder/cactus-test-framework/cactus-test-framework/workspace/plato-cactus-hibench_AQA-1256/hibench/bin/functions/load_config.py", line 723, in <module>
    load_config(conf_root, workload_configFile, workload_folder, patching_config)
  File "/home/users/builder/cactus-test-framework/cactus-test-framework/workspace/plato-cactus-hibench_AQA-1256/hibench/bin/functions/load_config.py", line 208, in load_config
    generate_optional_value()
  File "/home/users/builder/cactus-test-framework/cactus-test-framework/workspace/plato-cactus-hibench_AQA-1256/hibench/bin/functions/load_config.py", line 646, in generate_optional_value
    probe_masters_slaves_hostnames()
  File "/home/users/builder/cactus-test-framework/cactus-test-framework/workspace/plato-cactus-hibench_AQA-1256/hibench/bin/functions/load_config.py", line 581, in probe_masters_slaves_hostnames
    HibenchConf['hibench.masters.hostnames'].split() +
KeyError: 'hibench.masters.hostnames'
/home/users/builder/cactus-test-framework/cactus-test-framework/workspace/plato-cactus-hibench_AQA-1256/hibench/bin/functions/workload-functions.sh: line 38: .: filename argument required
.: usage: . filename [arguments]
start HadoopSleep bench
/home/users/builder/cactus-test-framework/cactus-test-framework/workspace/plato-cactus-hibench_AQA-1256/hibench/bin/workloads/micro/sleep/hadoop/run.sh: line 27: HADOOP_SLEEP_JAR: unbound variable
ERROR: micro/sleep/hadoop failed to run successfully.
```

**Comment by harschware** [comment]
my issue may be different from this one so I filed #363

**Comment by jiaqianjing** [comment]
@harschware Hi,have you ever been set configuration about  `mapreduce.home` in your HiBench 6.0.In the 5.0 release, I added the following in the file

>  99-user_defined_properties.conf

:
```
Hibench.hadoop.mapreduce.home /usr/hdp/2.4.0.0-169/hadoop-mapreduce
Hibench.hadoop.release hdp
```

**Comment by carsonwang** [comment]
@jiaqianjing , can you please show us your `hadoop.conf`? What is the path of `hadoop-mapreduce-examples.jar` for HDP2.4? Is it </YOUR/HADOOP/HOME>/hadoop-mapreduce-examples.jar?

**Comment by carsonwang** [comment]
@jiaqianjing , this seems to be a bug in 6.0. We'll check this for HDP.

**Comment by jiaqianjing** [comment]
Hi,@carsonwang , there is my configuration file:

> # Hadoop home
hibench.hadoop.home     /usr/hdp/2.4.0.0-169/hadoop

# The path of hadoop executable
hibench.hadoop.executable     ${hibench.hadoop.home}/bin/hadoop

# Hadoop configraution directory
hibench.hadoop.configure.dir  ${hibench.hadoop.home}/etc/hadoop

# The root HDFS path to store HiBench data
hibench.hdfs.master       hdfs://hdfs-ha


# Hadoop release provider. Supported value: apache, cdh5, hdp
hibench.hadoop.release    hdp

#Hadoop version. Only hadoop2 is supported
#TODO: remove this
hibench.hadoop.version    hadoop2

**Comment by jiaqianjing** [comment]
hi, @carsonwang  

> # Hadoop home
>hibench.hadoop.home     /usr/hdp/2.4.0.0-169/hadoop
># The path of hadoop executable
>hibench.hadoop.executable     ${hibench.hadoop.home}/bin/hadoop
># Hadoop configraution directory
>hibench.hadoop.configure.dir  ${hibench.hadoop.home}/etc/hadoop
># The root HDFS path to store HiBench data
>hibench.hdfs.master       hdfs://hdfs-ha
># Hadoop release provider. Supported value: apache, cdh5, hdp
>hibench.hadoop.release    hdp
>#Hadoop version. Only hadoop2 is supported
>#TODO: remove this
>hibench.hadoop.version    hadoop2

**Comment by jiaqianjing** [comment]
hi, @carsonwang  this is my configuration file about hadoop.conf:
`# Hadoop home
hibench.hadoop.home     /usr/hdp/2.4.0.0-169/hadoop

# The path of hadoop executable
hibench.hadoop.executable     ${hibench.hadoop.home}/bin/hadoop

# Hadoop configraution directory
hibench.hadoop.configure.dir  ${hibench.hadoop.home}/etc/hadoop

# The root HDFS path to store HiBench data
hibench.hdfs.master       hdfs://hdfs-ha


# Hadoop release provider. Supported value: apache, cdh5, hdp
hibench.hadoop.release    hdp

#Hadoop version. Only hadoop2 is supported
#TODO: remove this
hibench.hadoop.version    hadoop2
`

**Comment by jiaqianjing** [comment]
@carsonwang ,yep, 
my `hadoop-mapreduce-examples.jar`   is  not in the `$HADOOP_HOME/` 
hdp have an special directory --hadoop-mapreduce.
In this directory,include  a lot of jar package，for instance,hadoop-mapreduce-examples.jar

**Comment by jiaqianjing** [comment]
@carsonwang . I found the cause of the problem. Looking at HiBench6.0_HOME/bin/functions/load_config.py at line 382,you will see `hadoop_examples_jar` of directory.

**Comment by carsonwang** [comment]
Thanks @jiaqianjing . Yes, we need update the path. Did you use some earlier HDP versions? Do you know if the example jar path is always the same?

**Comment by jiaqianjing** [comment]
方便加个微信吗 @carsonwang ID:jiaqianjing123

**Comment by harschware** [comment]
I added this to my hadoop.conf file
```
hibench.hadoop.examples.jar /usr/hdp/current/hadoop-mapreduce-client/hadoop-mapreduce-examples.jar
hibench.hadoop.examples.test.jar /usr/hdp/current/hadoop-mapreduce-client/hadoop-mapreduce-client-jobclient.jar
```

**Comment by jiaqianjing** [comment]
@harschware  thx. Howerver,I have fixed it in HiBench6.0_HOME/bin/functions/load_config.py at line 382

**Comment by gczsjdy** [comment]
@jiaqianjing  How do you fix it?

**Comment by jiaqianjing** [comment]
@gczsjdy I just repaired for my directory structure. As you see:
`def probe_hadoop_examples_jars():
    # probe hadoop example jars
    if not HibenchConf.get("hibench.hadoop.examples.jar", ""):
        examples_jars_candidate_apache0 = HibenchConf[
            'hibench.hadoop.home'] + "/share/hadoop/mapreduce/hadoop-mapreduce-examples-*.jar"
        examples_jars_candidate_cdh0 = HibenchConf[
            'hibench.hadoop.home'] + "/share/hadoop/mapreduce2/hadoop-mapreduce-examples-*.jar"
        examples_jars_candidate_cdh1 = HibenchConf[
            'hibench.hadoop.home'] + "/../../jars/hadoop-mapreduce-examples-*.jar"
        examples_jars_candidate_hdp0 = HibenchConf[
            'hibench.hadoop.home'] + "/../hadoop-mapreduce/hadoop-mapreduce-examples.jar"
        examples_jars_candidate_list = [
            examples_jars_candidate_apache0,
            examples_jars_candidate_cdh0,
            examples_jars_candidate_cdh1,
            examples_jars_candidate_hdp0]

        HibenchConf["hibench.hadoop.examples.jar"] = exactly_one_file(
            examples_jars_candidate_list, "hibench.hadoop.examples.jar")
        HibenchConfRef["hibench.hadoop.examples.jar"] = "Inferred by " + \
            HibenchConf["hibench.hadoop.examples.jar"]
`

**Comment by gczsjdy** [comment]
@jiaqianjing Thanks, by the way, did you install HDP with Ambari?

**Comment by gczsjdy** [comment]
@harschware  Is your HDP home /usr/hdp or /usr/hdp/current?

**Comment by jiaqianjing** [comment]
@gczsjdy Yep, I used Ambari , but didn't modify the probing code. And my $HDP_HOME is /usr/hdp/VERSION/...

**Comment by carsonwang** [comment]
Resolved by #373
